src/main.rs

```diff
@@ -20,11 +20,39 @@ use zellij_utils::{
     structopt::StructOpt,
 };
 
+fn start_new_session_name(opts: &CliArgs) -> String {
```

Comment:
But session_name might be empty. I'll think about something.

Reply:
If the session_name is empty, the parameter `session_name` will be `None`. So, you can then operate on it the same way you do now:
```rust
fn start_new_session_name(name: Option<&String>) -> String {
    let session_name = name
        .map_or_else(|| names::Generator::default().next().unwrap(), |s| s.clone());
    assert_session_ne(&session_name);
    return session_name;
}
```

Reply:
I tried to get this working in various ways, but always every avenue I took ended up in failures. I guess I don't yet fully master the Rust ownership and types. Could you please show me what exactly I should have in the calling function?

Reply:
Please see https://github.com/kunalmohan/zellij/commit/ec464a2f539b9fdf6d13e086420e034091c0fecb for the relevant changes. Feel free to reach out in case there's any confusion.